### PR TITLE
upgrade to typescript 2.6.2

### PR DESCRIPTION
**Pull request by giladgray**
compiler now detects variables that are assigned but never read!

### CODE REVIEW

File: packages/table-dev-app/src/mutableTable.tsx [hunk 1]
```diff
@@ -106,7 +106,7 @@ const LONG_TEXT_WORD_SPLIT_REGEXP = /.{1,5}/g;
 const LARGE_JSON_PROP_COUNT = 3;
 const LARGE_JSON_OBJECT_DEPTH = 2;
 
-const CELL_CONTENT_GENERATORS = {
+const CELL_CONTENT_GENERATORS: { [name: string]: (ri: number, ci: number) => string | object } = {
```
Comment (gscshoyru) [inline packages/table-dev-app/src/mutableTable.tsx]: Make an interface for this maybe? Or some piece of it?

Comment (giladgray) [reply packages/table-dev-app/src/mutableTable.tsx]: meh it's just dev code, used exactly once

File: packages/datetime/src/dateInput.tsx [hunk 2]
```diff
@@ -451,12 +448,6 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
         this.safeInvokeInputProp("onKeyDown", e);
     };
 
-    private setInputRef = (el: HTMLElement) => {
-        this.inputRef = el;
-        const { inputProps = {} } = this.props;
-        Utils.safeInvoke(inputProps.inputRef, el);
```
Comment (adidahiya) [inline packages/datetime/src/dateInput.tsx]: why is this ok to remove? it's a documented feature

Comment (giladgray) [reply packages/datetime/src/dateInput.tsx]: I removed the unused `inputRef` (due to new compiler error -- it was never read), so `inputProps.inputRef` is applied from the `{...htmlInputProps}` spread

**Pull request merged**